### PR TITLE
chore: Mild reorganization of project files for better dev experience, tidiness

### LINKED ISSUE #64: chore: Mild reorganization of project files for better dev experience, tidiness
Self assigned, if the repo owner gives permission.

I'd like to move *wiktionaryparser.py* and the *utils.py* into a new *wiktionaryparser* folder, making the project structure conform to the structures of other Python repositories.

proj-structure

proj-structure-2

I'll make sure to add a *CONTRIBUTING.md* with the shell script contributors can copy paste to get their dev env setup (`venv`, `pip install -e .`). 

I'll also make sure to add the classes in *utils.py* to `__all__` so the API doesn't change.

The aim's just to:
- Make the repo tidier.
- Make Intellisense work again since, for some reason, the current project structure is preventing it from doing autocompletion.
- Pave the way for absolute imports.

### PR DESCRIPTION
**Pull request by sehwol**
Resolves #64 

## Changes

- Moved the files around and double checked that the API remains the same by installing it in a test project (see below).

api-is-still-the-same

- Added a *CONTRIBUTING.md* with specific, easy to copy instructions to get dev env set up. Successfully tested the instructions to clone into a separate folder:

```bash
# --single-branch and --branch i64 does not exist in the actual CONTRIBUTING.md
git clone https://github.com/sehwol/WiktionaryParser --single-branch --branch i64 && \
  cd WiktionaryParser && \
  virtualenv venv && \ 
  source ./venv/scripts/activate && \
  pip install -r requirements.txt && \
  pip install -e .

```

- Add missing dependencies to *requirements.txt*.
- Switch to absolute imports, conforming with style of other repos and fixing intellisense.
- Update *.gitattributes* so the repo would **hopefully** be treated as a Python repo and not an HTML repo. I say hopefully because though I'm following GitHub's own [instructions](https://github.com/github/linguist#overrides), it doesn't seem to be changing the lang of my forked repo.

### CODE REVIEW

File: scripts/download_wiktionary_test_files.py [hunk 1]
```diff
@@ -6,7 +6,8 @@
 import requests
 import os
 from typing import List, Set
-from tests import test
+# pylint: disable=F0401
+from tests import test_core as test
```
Comment (Suyash458) [inline scripts/download_wiktionary_test_files.py:10]: Any reason why `pylint` has issues with this particular import?

Comment (sehwol) [reply scripts/download_wiktionary_test_files.py:10]: I'm unsure. That script works when run in virtualenv and Visual Studio Code intellisense can travel to the import no problem. Only pylint's the one complaining. I'm currently looking if adding it in a new *.env* file would convince pylint that there's no problem.

**Comment by Suyash458** [comment]
This looks great, thanks for adding `CONTRIBUTING.md` and `.gitattributes` as well! I noticed the repo was being recognized as a HTML one and was about to mark them as `linguist-vendored`. The language classification is probably something that runs as a job and is not real time (there's a similar issue [here](https://github.com/github/linguist/issues/197)) so, it'll probably be updated in a while

File: scripts/download_wiktionary_test_files.py (hunk 1 above)
Comment (sehwol) [reply scripts/download_wiktionary_test_files.py:10]: > 
> 
> Any reason why `pylint` has issues with this particular import?

Okay, I just got rid of the comment and even removed the *.env* file with the following content
```
PYTHONPATH=tests/
```
Ran pylint again, and the complaint disappeared. Not sure what that's about.

Comment (sehwol) [reply scripts/download_wiktionary_test_files.py:10]: Will this issue prevent the accepting of this pull request?

Comment (Suyash458) [reply scripts/download_wiktionary_test_files.py:10]: Seems like this issue turns up if `pylint` is not installed in the same directory as python packages
https://github.com/Microsoft/vscode-python/issues/1185

Comment (sehwol) [reply scripts/download_wiktionary_test_files.py:10]: Odd, could have sworn that I already set VS to use the *python.exe* in venv. Want me to remove it?

Comment (Suyash458) [reply scripts/download_wiktionary_test_files.py:10]: > Will this issue prevent the accepting of this pull request?

Not really, I was just curious as to why `pylint` as having a problem with this specific import. I think this comment can be removed then?

Comment (sehwol) [reply scripts/download_wiktionary_test_files.py:9]: Got it. Comment removed and commits rebased.

Comment (Suyash458) [reply scripts/download_wiktionary_test_files.py:9]: > Odd, could have sworn that I already set VS to use the python.exe in venv

Yeah I've faced a fair share of weird VSCode issues as well hahah

**Pull request merged**